pkg/knowledge_base2/properties/int_property.go

```diff
@@ -65,6 +66,17 @@ func (i *IntProperty) Parse(value any, ctx knowledgebase.DynamicContext, data kn
 	if val, ok := value.(int); ok {
 		return val, nil
 	}
+	if val, ok := value.(float32); ok {
+		if float64(val) == math.Trunc(float64(val)) {
```

Comment:
Like I mentioned in the general comment, `==` is not a safe operation on floating-point numbers.

Reply:
out of curiousity, why isnt this a safe operation?

Reply:
Because floating point numbers are weird and in some circumstances have multiple ways of representing the same number. They can have rounding errors that could be incurred during parsing from string or any operations such as via template functions.

Reply:
but wouldnmt the approach of checking if  abs(float64(ival) - val) > EPSILON  have some similar errors in terms, ill implement the other route though

Reply:
It's to cover the case that, for example, `0.4+0.4+0.2 == 0.9999999`. By using an EPSILON instead of `==`, it would allow for small amounts of accumulated errors.